scripts/busybox-ping-test/remote_aca_build.sh

```diff
@@ -0,0 +1,86 @@
+#! /bin/sh
+
+# Build ACA on ACA nodes from alcor_services.ini
+# Triggers a remote build on ACA nodes listed in alcor_services.ini.
+#
+
+ALCOR_INI=alcor_services.ini
+SCRIPT_DIR=`dirname $0`
+SCRIPT_DIR=`realpath $SCRIPT_DIR`
+
+ACA_DIR="repos/aca"
+ACA_NODES=""
+
+ACA_REPO="futurewei-cloud"
+ACA_BRANCH="master"
+ACA_COMMIT="HEAD"
+USER_REPO=0
+USER_BRANCH=0
+
+get_aca_node_info() {
+     ACA_NODES=`sed -n '/^\[AlcorControlAgents\]/,/^\[/p' ${ALCOR_INI} | grep -v '^\[' | grep -v '^[\t ]*$' |  awk -F= '{print $2}' | tr -d '[\t ]'`
+}
+
+if [ -d "${SCRIPT_DIR}" -a -s ${SCRIPT_DIR}/${ALCOR_INI} ]; then
+    true
+else
+    echo "Missing ${ALCOR_INI} file, can only run in alcor/scripts/busybox-ping-test"
+    exit 1
+fi
+
+while getopts "r:b:c:f" opt; do
+    case "$opt" in
+        r) ACA_REPO=$OPTARG
+            USER_REPO=1
+            ;;
+
+        b) ACA_BRANCH=$OPTARG
+            USER_BRANCH=1
+            ;;
+
+        c) ACA_COMMIT=$OPTARG
+            ;;
+
+        f)  DO_FORCE=1
+            ;;
+        ?)
+            echo "Usage: $0 [-r repo] [-b branch] [-c commit] [-f]
+-r repo     : repository, if using a personal fork.
+-b branch   : branch, if using a personal branch.
+-c commit   : commit, if not using the HEAD.
+-f          : to force a build even if there are no changes in remote.
+"
+            exit 1
+            ;;
+    esac
+done
+
+get_aca_node_info
+NC=0
+for node in `echo ${ACA_NODES}`; do
+    rm -f /tmp/aca_${node}_build.log > /dev/null 2>&1
+    NC=`expr $NC + 1`
+    scp ./build_aca.sh ubuntu@${node}:${ACA_DIR}/
+    ssh ubuntu@$node "cd $ACA_DIR && sudo ./build_aca.sh $ACA_REPO $ACA_BRANCH $ACA_COMMIT $DO_FORCE" > /tmp/aca_${node}_build.log 2>&1 &
+    echo "Scheduled build on $node"
+done
+
+# check the status
+SC=0
```

Comment:
Do we have a dummy API in ACA for us to verify it is up?

Reply:
No. one or more of ps output, netstat output, and possibly trawling through syslog allow us to learn if ACA is running or not.